### PR TITLE
azurerm_hdinsight_*_cluster - fix broken updaten due to api not allowing cluster to be disabled

### PR DESCRIPTION
**Pull request by kosinsky**
Fixes issues like https://github.com/terraform-providers/terraform-provider-azurerm/issues/7067

HDInisght doesn't support disabling Gateway/Ambari for Linux clusters and only Linux clusters supported. 
Making gateway.enabled property always equal true.
Also using API call to update Gateway settings (changing password)

### CODE REVIEW

**Comment by magodo** [comment]
@kosinsky Thanks for updating the code!
@katbyte Would you please take a look at this PR? It looks good to me. Besides, there is another PR (#7272 ) that is related with #7067 you might also want to review.

File: website/docs/r/hdinsight_hadoop_cluster.html.markdown [hunk 1]
```diff
@@ -123,9 +123,9 @@ A `component_version` block supports the following:
 
 A `gateway` block supports the following:
 
-* `enabled` - (Required) Is the Ambari portal enabled? Changing this forces a new resource to be created.
+* `enabled` - (Optional) Is the Ambari portal enabled? Deprecated: HDInsight doesn't support disabling gateway anymore.
```
Comment (katbyte) [inline website/docs/r/hdinsight_hadoop_cluster.html.markdown:126]:
```suggestion
* `enabled` - (Optional/ **Deprecated) Is the Ambari portal enabled? The HDInsight API doesn't support disabling gateway anymore.
```

File: website/docs/r/hdinsight_hbase_cluster.html.markdown [hunk 2]
```diff
@@ -121,9 +121,9 @@ A `component_version` block supports the following:
 
 A `gateway` block supports the following:
 
-* `enabled` - (Required) Is the Ambari portal enabled? Changing this forces a new resource to be created.
+* `enabled` - (Optional) Is the Ambari portal enabled? Deprecated: HDInsight doesn't support disabling gateway anymore.
```
Comment (katbyte) [inline website/docs/r/hdinsight_hbase_cluster.html.markdown:124]:
```suggestion
* `enabled` - (Optional/ **Deprecated) Is the Ambari portal enabled? The HDInsight API doesn't support disabling gateway anymore.
```

File: website/docs/r/hdinsight_interactive_query_cluster.html.markdown [hunk 3]
```diff
@@ -119,9 +119,9 @@ A `component_version` block supports the following:
 
 A `gateway` block supports the following:
 
-* `enabled` - (Required) Is the Ambari portal enabled? Changing this forces a new resource to be created.
+* `enabled` - (Optional) Is the Ambari portal enabled? Deprecated: HDInsight doesn't support disabling gateway anymore.
```
Comment (katbyte) [inline website/docs/r/hdinsight_interactive_query_cluster.html.markdown:122]:
```suggestion
* `enabled` - (Optional/ **Deprecated) Is the Ambari portal enabled? The HDInsight API doesn't support disabling gateway anymore.
```

File: website/docs/r/hdinsight_kafka_cluster.html.markdown [hunk 4]
```diff
@@ -122,9 +122,9 @@ A `component_version` block supports the following:
 
 A `gateway` block supports the following:
 
-* `enabled` - (Required) Is the Ambari portal enabled? Changing this forces a new resource to be created.
+* `enabled` - (Optional) Is the Ambari portal enabled? Deprecated: HDInsight doesn't support disabling gateway anymore.
```
Comment (katbyte) [inline website/docs/r/hdinsight_kafka_cluster.html.markdown:125]:
```suggestion
* `enabled` - (Optional/ **Deprecated) Is the Ambari portal enabled? The HDInsight API doesn't support disabling gateway anymore.
```

File: website/docs/r/hdinsight_ml_services_cluster.html.markdown [hunk 5]
```diff
@@ -116,9 +116,9 @@ The following arguments are supported:
 
 A `gateway` block supports the following:
 
-* `enabled` - (Required) Is the Ambari portal enabled? Changing this forces a new resource to be created.
+* `enabled` - (Optional) Is the Ambari portal enabled? Deprecated: HDInsight doesn't support disabling gateway anymore.
```
Comment (katbyte) [inline website/docs/r/hdinsight_ml_services_cluster.html.markdown:119]:
```suggestion
* `enabled` - (Optional/ **Deprecated) Is the Ambari portal enabled? The HDInsight API doesn't support disabling gateway anymore.
```

File: website/docs/r/hdinsight_rserver_cluster.html.markdown [hunk 6]
```diff
@@ -116,9 +116,9 @@ The following arguments are supported:
 
 A `gateway` block supports the following:
 
-* `enabled` - (Required) Is the Ambari portal enabled? Changing this forces a new resource to be created.
+* `enabled` - (Optional) Is the Ambari portal enabled? Deprecated: HDInsight doesn't support disabling gateway anymore.
```
Comment (katbyte) [inline website/docs/r/hdinsight_rserver_cluster.html.markdown:119]:
```suggestion
* `enabled` - (Optional/ **Deprecated) Is the Ambari portal enabled? The HDInsight API doesn't support disabling gateway anymore.
```

File: website/docs/r/hdinsight_spark_cluster.html.markdown [hunk 7]
```diff
@@ -121,9 +121,9 @@ A `component_version` block supports the following:
 
 A `gateway` block supports the following:
 
-* `enabled` - (Required) Is the Ambari portal enabled? Changing this forces a new resource to be created.
+* `enabled` - (Optional) Is the Ambari portal enabled? Deprecated: HDInsight doesn't support disabling gateway anymore.
```
Comment (katbyte) [inline website/docs/r/hdinsight_spark_cluster.html.markdown:124]:
```suggestion
* `enabled` - (Optional/ **Deprecated) Is the Ambari portal enabled? The HDInsight API doesn't support disabling gateway anymore.
```

File: website/docs/r/hdinsight_storm_cluster.html.markdown [hunk 8]
```diff
@@ -119,9 +119,9 @@ A `component_version` block supports the following:
 
 A `gateway` block supports the following:
 
-* `enabled` - (Required) Is the Ambari portal enabled? Changing this forces a new resource to be created.
+* `enabled` - (Optional) Is the Ambari portal enabled? Deprecated: HDInsight doesn't support disabling gateway anymore.
```
Comment (katbyte) [inline website/docs/r/hdinsight_storm_cluster.html.markdown:122]:
```suggestion
* `enabled` - (Optional/ **Deprecated) Is the Ambari portal enabled? The HDInsight API doesn't support disabling gateway anymore.
```

**Pull request merged**